mantidproject/mantid
# Move GitHub api keys

As described in #30357, github invalidated our API keys, this moves how the keys are shared and inserted into packages. Additionally, this allows for overriding the key via `ConfigService`.

**To test:**

The main way is to turn on debug logging, then run `DownloadInstrument` after configuring `cmake -DGITHUB_AUTHORIZATION_TOKEN=9f1c1acd61ecb87b21ad0382f6b403c8294992c5 .` (rejected old key). This will get the old authorization error back, and fall back to unauthenticated. After that, set `network.github.api_token=unset` in your properties file and see that it only tries unauthenticated access.

Fixes #30356 
Fixes #30357

The version into `ornl-next` is #30565

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
